internal/app/orchestrator/orchestrator.go

```diff
@@ -187,13 +206,34 @@ func (o *orchestrator) CreateWatch(req transport.Request) (transport.Watch, func
 				metrics.OrchestratorWatchErrorsSubscope(o.scope, aggregatedKey).Counter(metrics.ErrorChannelFull).Inc(1)
 			} else {
 				metrics.OrchestratorWatchSubscope(o.scope, aggregatedKey).Counter(metrics.OrchestratorWatchFanouts).Inc(1)
+				if isNackRequest {
+					o.logger.With(
+						"aggregated_key", aggregatedKey,
+						"request_version", req.GetVersionInfo(),
+						"response_version", cached.Resp.GetPayloadVersion(),
+						"request_type", req.GetTypeURL(),
+						"error", req.GetError(),
+						"resource_names", resourceString,
+					).Debug(context.Background(), "NACK request receives cached response with different version")
+				}
 			}
 		}()
 	}
 
 	// Check if we have a upstream stream open for this aggregated key. If not,
 	// open a stream with the representative request.
 	if !o.upstreamResponseMap.exists(aggregatedKey) {
+		// Should not happen, but logging behavior as a sanity check
+		if isNackRequest {
```

Comment:
do we need to log again?

Reply:
not really, it was more for me to have easy messages to read that show how a NACK request moves through the CreateWatch flow. i went ahead and removed the redundant ones